modules/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/codegen/CodeGenerator.java

```diff
@@ -3085,4 +3111,55 @@ private int getTypeCPIndex(BLangExpression expr) {
         TypeRefCPEntry typeRefCPEntry = new TypeRefCPEntry(typeSigCPIndex);
         return currentPkgInfo.addCPEntry(typeRefCPEntry);
     }
+
+    private int getValueToRefTypeCastOpcode(int typeTag) {
```

Comment:
There s an exception here. For ref types we simply need to return a NOP instruction. I2ANY, F2ANY ... does not have ref instruction where when we add ref offset it will return ref opcode. And I2ANY like cast operators are resolved at semantic phase, in our code we have not used those with getOpcode(). I am not sure whether we can use this method without any modification or whether we have a better way to handle this.